src/CoreWCF.NetTcp/src/CoreWCF/NetTcpBinding.cs

```diff
@@ -135,17 +148,37 @@ private void CheckSettings()
             {
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.UnsupportedSecuritySetting, "Mode", mode)));
             }
-
-            // Message.ClientCredentialType = Certificate, IssuedToken or Windows are not supported.
             if (mode == SecurityMode.TransportWithMessageCredential)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.UnsupportedSecuritySetting, "Mode", mode)));
+                MessageSecurityOverTcp message = security.Message;
+                if (message != null)
+                {
+                    MessageCredentialType mct = message.ClientCredentialType;
+                    if ((mct == MessageCredentialType.Certificate) || (mct == MessageCredentialType.IssuedToken) || (mct == MessageCredentialType.Windows))
+                    {
+                        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.UnsupportedSecuritySetting, "Message.ClientCredentialType", mct)));
+                    }
+                }
             }
+
+            // Transport.ClientCredentialType = Certificate is not supported.
+            Fx.Assert((mode == SecurityMode.Transport) || (mode == SecurityMode.TransportWithMessageCredential), "Unexpected SecurityMode value: " + mode);
+            TcpTransportSecurity transport = security.Transport;
+            if ((transport != null) && (transport.ClientCredentialType == TcpClientCredentialType.Certificate))
+            {
```

Comment:
Why is certificate credential type with transport security resulting in throwing?

Reply:
I think this was specific check for older OS version.. I have removed CheckSettings().. refer the call here https://referencesource.microsoft.com/#System.ServiceModel/System/ServiceModel/Channels/UnsafeNativeMethods.cs,1162